### DISCUSSION TITLE
Setting runlevel at kernel cmdline vs config

### DISCUSSION THREAD

**Discussion by hongkongkiwi** [finit-project/finit#259] [Q&A]
I'm wondering the read order of runlevel in finit (cmdline/config).

Right now, I have runlevel 3 set in my /etc/finit.conf config via `runlevel 3`. This makes sense as a default.

If the runlevel is set via the kernel commandline I would like to override the runlevel set in the config (because obviously we are specifically asking for it to be something). That's how I expected it to work but it seems to not be doing that.

Is this a bug or by design?

e.g. here's my kernel cmdline:
`quiet 9 -- finit.fstab=/etc/myfstab`

So according to this I should expect finit will launch at runlevel 9 , but after reading the finit.conf it changes it to runlevel 3.

**Comment by hongkongkiwi** [comment]
I see there is a `--with-runlevel` option.

It seems like the solution is to remove runlevel from my config and set the default to 3 via this compile time option, then the cmdline should override.

This was a bit confusing as I expected kernel to always take priority if set.

**Reply by hongkongkiwi** [reply]
After testing, I can't find anyway to set the runlevel using kernel cmdline.
1. Removed runlevel config in /etc/finit.conf
2. Set compile time option for runlevel to 3
3. Set kernel cmdline: `quiet  9 -- finit.fstab=/etc/myfstab 8`
(note: I tried both methods, setting before -- and after just in case)

After booting, runlevel is still 3 when I check with initctl runlevel

**Reply by troglobit** [reply]
I think the original intention, way back, was to support runlevel from the kernel command line.  The somewhere along the line support for it was lost or garbled.  The currently supported kernel command line options are documented in `doc/cmdline.md`.

The `--with-runlevel` is just the default runlevel to start with, unless one is defined in `/etc/finit.conf`.

I'll create an issue to deal with this situation.  Like you, I also think that the command line should override.  Let me see what I can do about it this weekend.

**Reply by hongkongkiwi** [reply]
Thanks! This will help a lot.

I can't set the runlevel during bootup either with a script because of the initctl limitations during startup. I made a simple script to do it before I remembered that limitation.
